### ISSUE TITLE
Pass/Fail status in 'Copy all as text' is not completely correct

### ISSUE THREAD

**Issue by lloydi** [superhighfives/pika#63]
This is regarding the 0.0.11-beta1 you just put out.

When choosing to 'Copy all as text', some of the data is incorrect. Attached image shows the issue (WCAG AA is marked as a pass when it is a fail. The key on the app UI is correctly showing pass/fail status.

<img width="1001" alt="pika_copy_all_as_text" src="https://user-images.githubusercontent.com/2778763/113264338-c3be2680-92ca-11eb-8881-2bda3ce55951.png">

Also, having seen the output as text, I think it could be made less ambiguous by not combining AA and AAA large as you do in the app UI. I would suggest that the format should be as follows (and using the example colors above to show pass/fails):

Foreground: #ececec / rgb(236, 236, 236) / hsb(360, 0, 93) / hsl(0, 0, 93)
Background: #7a7a7a / rgb(122, 122, 122) / hsb(120, 0, 48) / hsl(120, 0, 48)
Contrast Ratio: 3.63:1
WCAG Text contrast:
* AA: Fail
* AA Large: Pass
* AAA: Fail
* AAA Large: Fail

Note that I am emphasizing that this is for *Text Contrast*. There is also an SC for Non-text items contrast. This could also potentially be included in this text output (as you do not have the limitations of a small space on the app Window for that.

[SC 1.4.3 Contrast (Minimum)](https://www.w3.org/TR/WCAG21/#contrast-minimum)
[SC 1.4.11 Non-text Contrast](https://www.w3.org/TR/WCAG21/#non-text-contrast)

**Comment by superhighfives** [comment]
Hey @lloydi—good catch on the wording for the text copy. Will resolve in the next beta. How would you include the non-text contrast information in the text copy? Currently I'm finding the text version a bit verbose and unwieldy, so would be keen to simplify it, perhaps by placing the pass / fail information on one line. For example:
WCAG Compliance: AA (fail) / AA Large (fail) / AAA (fail) / AAA Large (fail) / Non-text (fail)

Thinking of the case where you have a couple of examples you're comparing.

**Comment by lloydi** [comment]
In my opinion, once you've copied all as text, there is not a big requirement to keep it brief (as you are forced to/want to in the app UI key), but more important to have it structured and understandable, hence going for bullet points rather than in one line.

So if you could also include Nontext contrast, I'd prefer if the structure were like this:

Foreground: #ececec / rgb(236, 236, 236) / hsb(360, 0, 93) / hsl(0, 0, 93)
Background: #7a7a7a / rgb(122, 122, 122) / hsb(120, 0, 48) / hsl(120, 0, 48)
Contrast Ratio: 3.63:1
WCAG Text contrast:
* AA: Fail
* AA Large: Pass
* AAA: Fail
* AAA Large: Fail
WCAG Non-Text contrast: Pass

As 1.4.11 is only set at AA level and there is no change in terms of size as with text so it's a simple pass/fail.

That said, I'd still be totally happy if it were output like this:

Foreground: #ececec / rgb(236, 236, 236) / hsb(360, 0, 93) / hsl(0, 0, 93)
Background: #7a7a7a / rgb(122, 122, 122) / hsb(120, 0, 48) / hsl(120, 0, 48)
Contrast Ratio: 3.63:1
WCAG Text contrast: AA: Fail / AA Large: Pass / AAA: Fail / AAA Large: Fail
WCAG Non-Text contrast: Pass

😀

**Comment by superhighfives** [comment]
Been noodling on this this morning, and I'm planning to go with the following for beta `0.0.11-beta2`.
```
Foreground: #0045ef · rgb(0, 69, 239) · hsb(223, 100, 94) · hsl(223, 100, 47)
Background: #000000 · rgb(0, 0, 0) · hsb(0, 0, 0) · hsl(0, 0, 0)
Contrast Ratio: 3.09:1
WCAG Compliance: Non-text (Pass) · AA Large (Pass) · AA / AAA Large (Fail) · AAA (Fail)
```

Thanks for all the feedback, and for the kind words. 🙌

**Comment by lloydi** [comment]
I would recommend a slight change:

WCAG Compliance: Text: AA Large (Pass) · AA / AAA Large (Fail) · AAA (Fail) Non-text: Pass

Or something like that. Reasons being:

* As it stands it could be read as Non-text is the label for all the text that follows 
* In terms of WCAG, Non-Text contrast (1.4.11) SC came in 2.1, after 1.4.3, so you could argue that 1.4.3 is a more pressing issue/more prevalent and therefore should come first in the list

**Comment by superhighfives** [comment]
Yeah, sounds good. Updated example, @lloydi:

```
Foreground: #bf225f · rgb(191, 34, 95) · hsb(336, 82, 75) · hsl(336, 70, 44)
Background: #000000 · rgb(0, 0, 0) · hsb(0, 0, 0) · hsl(0, 0, 0)
Contrast Ratio: 3.6:1
WCAG Compliance: AA Large (Pass) · AA / AAA Large (Fail) · AAA (Fail) · Non-text (Pass)
```

**Comment by lloydi** [comment]
Smashin'

**Comment by superhighfives** [comment]
Great, this is now available in `0.0.11-beta2`. You can either [download it here](https://github.com/superhighfives/pika/releases/tag/0.0.11-beta2) or enable beta's in the app preferences and check for updates. 👍

**Comment by lloydi** [comment]
Lovely stuff

**Comment by superhighfives** [comment]
Awesome, this is now live in v0.0.11. 🥳 https://github.com/superhighfives/pika/releases/tag/0.0.11
